src/flow_solver/flow_solver_cases/periodic_turbulence.cpp

```diff
@@ -198,6 +199,103 @@ double PeriodicTurbulence<dim, nstate>::get_deviatoric_strain_rate_tensor_based_
+
+    OPERATOR::basis_functions<dim,2*dim> soln_basis(1, poly_degree, dg->max_grid_degree); 
+    soln_basis.build_1D_volume_operator(dg->oneD_fe_collection_1state[poly_degree], dg->oneD_quadrature_collection[poly_degree]);
+
+    dealii::LinearAlgebra::distributed::Vector<double> entropy_var_hat_global(dg->right_hand_side);
+    std::vector<dealii::types::global_dof_index> dofs_indices (n_dofs_cell);
+
+    //std::shared_ptr < Physics::PhysicsBase<dim, nstate, double > > pde_physics_double  = PHiLiP::Physics::PhysicsFactory<dim,nstate,double>::create_Physics(dg->all_parameters);
+
+    for (auto cell = dg->dof_handler.begin_active(); cell!=dg->dof_handler.end(); ++cell) {
+        if (!cell->is_locally_owned()) continue;
+        cell->get_dof_indices (dofs_indices);
+
+        std::array<std::vector<double>,nstate> soln_coeff;
+        for(unsigned int idof=0; idof<n_dofs_cell; idof++){
+            const unsigned int istate = dg->fe_collection[poly_degree].system_to_component_index(idof).first;
+            const unsigned int ishape = dg->fe_collection[poly_degree].system_to_component_index(idof).second;
+            if(ishape == 0)
+                soln_coeff[istate].resize(n_shape_fns);
+            soln_coeff[istate][ishape] = dg->solution(dofs_indices[idof]);
+        }
+
+        std::array<std::vector<double>,nstate> soln_at_q;
+        for(int istate=0; istate<nstate; istate++){
+            soln_at_q[istate].resize(n_quad_pts);
+            soln_basis.matrix_vector_mult_1D(soln_coeff[istate], soln_at_q[istate],
+                                             soln_basis.oneD_vol_operator);
+        }
+        std::array<std::vector<double>,nstate> entropy_var_at_q;
+        for(unsigned int iquad=0; iquad<n_quad_pts; iquad++){
+            std::array<double,nstate> soln_state;
+            for(int istate=0; istate<nstate; istate++){
+                soln_state[istate] = soln_at_q[istate][iquad];
+            }
+
+            std::array<double,nstate> entropy_var;
+            const double density = soln_state[0];
+            dealii::Tensor<1,dim,double> vel;
+            double vel2 = 0.0;
+            for(int idim=0; idim<dim; idim++){
+                vel[idim] = soln_state[idim+1]/soln_state[0];
+                vel2 += vel[idim]*vel[idim];
+            }
+            const double pressure = 0.4*(soln_state[nstate-1] - 0.5*density*vel2);
+            const double entropy = log(pressure) - 1.4 * log(density);
+            
+            //pcout << pressure << " " << entropy << std::endl;        
+            entropy_var[0] = (1.4-entropy)/0.4 - 0.5 * density / pressure * vel2;
+            for(int idim=0; idim<dim; idim++){
+                entropy_var[idim+1] = soln_state[idim+1] / pressure;
+            }
+            entropy_var[nstate-1] = - density / pressure;
+            //pcout << entropy_var[0] << " ";
```

Comment:
use a PhysicsBase object (`this->navier_stokes_physics`) for computing physics-related quantities

Reply:
I am going to add a `compute_entropy_variables()` function to physics on my RRK branch, which will enable me to use Physics here. I wanted to keep the changes on this branch minimal, hence the hard-coded physics.

Reply:
Added a function `convert_conservative_to_entropy()` to Euler. The test runs the same as before.